### ISSUE TITLE
Remove `check_species()`, `get_scientific_name()`

### ISSUE THREAD

**Issue by peterdesmet** [inbo/camtraptor#235]
_Suggested in camtraptor July 2023 coding sprint_

The functions `get_scientific_name()` returns a vector of scientific names for provided vernacular names, within the scope of a package. It does not allow scientific name values:

```R
> get_scientific_name(mica, c("bever", "mens"))
[1] "Castor fiber" "Homo sapiens"

> get_scientific_name(mica, c("bever", "mens", "Ardea")
+ )
Error: Invalid value for vernacular_name parameter: ardea.
Valid inputs are: mallard, gadwall, great herons, grey heron, eurasian beaver, human, beech marten, european polecat, red fox, wilde eend, krakeend, reigers, blauwe reiger, bever, mens, steenmarter, bunzing and vos.
```

The function `check_species()` returns a vector of scientific names for provided vernacular names _and_ scientific names, within the scope of a package:

```R
> check_species(mica, c("bever", "mens", "ardea"))
Scientific name of bever: Castor fiber
Scientific name of mens: Homo sapiens
[1] "Castor fiber" "Homo sapiens" "Ardea" 
```

We think these functions are no longer necessary:

- The current functions are confusing (what is their difference?)
- The current functions translate vernacular names to scientific names, but only based on the info that is available in a package. So one could not get a result because the species is not present or (worse) the species is present, but the vernacular name is not included.
- The suggested `taxonomy()` inbo/camtraptor#115 function provides an overview of taxa that are present in a (filtered) package in a **tabular format**. That information is easier to scan for existing (and missing) vernacular names that the current functions.
- The suggested `filter_` functions inbo/camtrapdp#23 allow to search on any column, e.g. `scientificName`, `vernacularName.nld` etc. Having functions that convert vernacular names to scientific names are less useful here.

**Comment by damianooldoni** [comment]
Agree. To be added to defunct functions.

**Comment by damianooldoni** [comment]
Implemented in `version-1.0` (dev branch).
